### PR TITLE
Add JavaScript testing of coverage creation

### PR DESCRIPTION
**Pull request by derekprior**
We will be adding some JavaScript to this page in order to support
adding many outcomes to a coverage dynamically. To get adequate test
coverage on this, we need to have a JavaScript test driver.

We usually use capybara-webkit for this, but QT was giving me troubles
and the world seems to be moving on to chromedriver. I decided to give
chromedriver a try here. Installation was an absolute picnic compared to
QT and capybara-webkit.

With the test case now running in a JavaScript driver, we can no longer
faithfully interact with the select boxes, which are actually hidden by
Selectize. I updated the test to interact with the Selectize input
instead.

### CODE REVIEW

File: spec/features/user_adds_coverage_to_a_course_spec.rb [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
+    selectize outcome.nickname, from: "Outcome"
     click_button t('helpers.submit.coverage.create')
 
     within("#matched_outcomes") do
       expect(page).to have_content(subject.title)
       expect(page).to have_content(outcome.nickname)
     end
   end
+
+  def selectize(item, from:)
```
Comment (derekprior) [inline spec/features/user_adds_coverage_to_a_course_spec.rb]: Is all of this too much? This is how I replicate selecting from a selectize'd input. Should I use selectize's javascript api instead (via execute_script)?

File: spec/features/user_adds_coverage_to_a_course_spec.rb [hunk 2]
```diff
@@ -1,21 +1,27 @@
...
+    container = find_field(from, visible: false).first(:xpath, ".//..")
```
Comment (calebthompson) [inline spec/features/user_adds_coverage_to_a_course_spec.rb]: visible: false scares me. Do we know that a user can click on it and it's not behind some div capturing clicks?

**Comment by calebthompson** [comment]
lgtm

File: spec/features/user_adds_coverage_to_a_course_spec.rb (hunk 1 above)
Comment (margaretoti) [inline spec/features/user_adds_coverage_to_a_course_spec.rb]: Perhaps using the selectize javascript api is a good idea. I foresee needing to use it to implement the nice form designs in the mockups. That said, I am not 100 % sure as I do not think that using selectize will necessarily make the code cleaner. From the documentation, it looks like you need to call quite a few functions. Maybe we can look at this together; I am curious to see what you had in mind if the selectize api is used.

File: spec/features/user_adds_coverage_to_a_course_spec.rb (hunk 2 above)
Comment (derekprior) [inline spec/features/user_adds_coverage_to_a_course_spec.rb]: The user can't click on it. This is the real `select` box, which `selectize` hides with it's fancy select box. In order to interact with the correct `selectize` box, I find the hidden select and then interact with visible elements that are in the same container.

File: spec/features/user_adds_coverage_to_a_course_spec.rb (hunk 1 above)
Comment (derekprior) [inline spec/features/user_adds_coverage_to_a_course_spec.rb]: I looked into this a bit, and I think the only things that would change would be the 2nd and 3rd lines of this method -- we'd be able to replace those with `page.driver.execute_script` executing some javascript to set the value of the selectize. It doesn't look like it'd be much simpler than what we have here and what we have here is a bit more true to how a user would interact with it. So I think we should go with this for now. Can revisit if this gives us problems.

**Pull request merged**